### PR TITLE
Improve Typescript typings

### PR DESCRIPTION
**Pull request by stackoversnow**
Lot of bad/unexpected/edge behaviors can be avoided when strongly typing.

Should prefer to steer away from force typing things with **!** as things are never as they may seem to be, even when they seem to be so the most 😉 .

### CODE REVIEW

File: src/app/home/home.component.ts [hunk 1]
```diff
@@ -139,16 +139,20 @@ export class HomeComponent implements AfterViewInit {
     }
 
     // TODO data-placeholder broke
-    onTextPaste($event: any) {
+    onTextPaste($event: ClipboardEvent) {
```
Comment (AndiBraimllari) [inline src/app/home/home.component.ts:142]: Can you double check if it is a `ClipboardEvent` regardless of if you paste through the keyboard or through the mouse?

Comment (stackoversnow) [reply src/app/home/home.component.ts:142]: Yes it works for both cases. There's no other option for this type of event. Also tested to double check 👍

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -374,15 +378,21 @@ export class HomeComponent implements AfterViewInit {
     }
 
 
-    getTextOfTextMarking(i: number) {
-        const editor = document.getElementById(this.EDITOR_KEY)!;
+    getTextOfTextMarking(i: number): string {
+        const editor: HTMLElement | null = document.getElementById(this.EDITOR_KEY)!;
```
Comment (AndiBraimllari) [inline src/app/home/home.component.ts:382]: I guess the `!` at the end of this line should be dropped

**Review by AndiBraimllari — CHANGES_REQUESTED** [review]

**Comment by AndiBraimllari** [comment]
You can also replace `''` with `this.EMPTY_STRING`

File: src/app/home/home.component.ts (hunk 2 above)
Comment (stackoversnow) [reply src/app/home/home.component.ts]: 👍

**Review by AndiBraimllari — APPROVED** [review]
Perhaps in the future in another PR we can replace the empty spaces with `null` e.g. at `this.markEditor(this.EMPTY_STRING, CursorPosition.END);`.
Approving this one, feel free to merge.

**Pull request merged**